aten/src/ATen/native/cuda/LegacyDefinitions.cu

```diff
@@ -0,0 +1,930 @@
+#include <ATen/ATen.h>
```

Comment:
I don't think there is any reason this should be a cu file. And it is a good idea to not put things in cu files if it is not necessary, since nvcc compiles quite a bit more slowly.

Reply:
It doesn't doesn't quite support all of C++.